src/lib.rs

```diff
@@ -285,7 +297,7 @@ impl<K: Hash + Eq, V, S: BuildHasher> LruCache<K, V, S> {
                 let node_ptr: *mut LruEntry<K, V> = &mut *node;
                 self.attach(node_ptr);
 
-                let keyref = unsafe { &(*node_ptr).key };
+                let keyref = unsafe { &(*node_ptr).kv.as_ref().unwrap().key };
```

Comment:
I think this particular `unsafe` is unnecessary:
```suggestion
                let keyref = &node.kv.as_ref().unwrap().key;
```